### DISCUSSION TITLE
Best way to contribute?

### DISCUSSION THREAD

**Discussion by GCuser99** [uezo/TinySeleniumVBA#48] [Q&A] [answered]
@uezo and others,

I'm a newbie to GitHub collaboration. I have been adding proposed code and object model ideas to the TinySeleniumVBA Issue bin, but I am wondering if there isn't a better way to do that, as many of the contributions there are not really issues but functionality suggestions. I apologize for helping to create what looks like a mess! Again, I am learning.

Is there a preferred way/place to contribute code suggestions and enhancements that really aren't "issues"?

Thanks,

GCUser99

**Comment by uezo** [comment, marked as answer]
@GCuser99 Thank you for asking me this very important issue. And, I apologize for late response and updating the codes.

Generally, the flow is like below:

1. Fork this repo and clone it to your machine.
1. Change/Add/Delete code.
1. Commit to your repo and push to GitHub.
1. Create pull request to this repo.
1. Review the pull request and merge it if it has no problems.

1-4 by contributor and just 5 by repository owner.

These flow works well for the general program languages like Python, C#, Java and so on because the source codes are the text files. However source code of VBA is included in the Excel file that can't be compared directly like text files. So you have to export or copy & paste the modified program every time you want to compare. This is a productivity issue I think.

If you can, give me pull request just adding comment to a module / class as training.
